### PR TITLE
[ButtonBar] add traitCollectionDidChangeBlock support.

### LINKED ISSUE #7979: [ButtonBar] Internal issue: b/137787145
This was filed as an internal issue. If you are a Googler, please visit [b/137787145](http://b/137787145) for more details.

---
#### Internal data
- Associated internal bug: [b/137787145](http://b/137787145)

### PR DESCRIPTION
**Pull request by fluxminds**
closes #7979

### CODE REVIEW

**Review by codeman7 — APPROVED** [review]

File: components/ButtonBar/tests/unit/ButtonBarTests.m [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialButtonBar.h"
+
+@interface ButtonBarTests : XCTestCase
+
+@property(nonatomic, strong, nullable) MDCButtonBar *buttonBar;
```
Comment (romoore) [inline components/ButtonBar/tests/unit/ButtonBarTests.m]: Nit: Please name this `MDCButtonBarTests` so Xcode will associate it with the header/implementation files.

File: components/ButtonBar/tests/unit/ButtonBarTests.m [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialButtonBar.h"
+
+@interface ButtonBarTests : XCTestCase
```
Comment (romoore) [inline components/ButtonBar/tests/unit/ButtonBarTests.m]: Nit: Please add documentation to the interface.

**Review by romoore — APPROVED** [review]

File: components/ButtonBar/tests/unit/ButtonBarTests.m (hunk 2 above)
Comment (fluxminds) [reply components/ButtonBar/tests/unit/ButtonBarTests.m]: Addressed in 62eaf4c

File: components/ButtonBar/tests/unit/ButtonBarTests.m (hunk 1 above)
Comment (fluxminds) [reply components/ButtonBar/tests/unit/ButtonBarTests.m]: Addressed in 62eaf4c

**Pull request merged**